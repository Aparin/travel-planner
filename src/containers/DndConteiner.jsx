import React, { useState } from 'react';
import update from 'immutability-helper';
import PlaceCard from './DndPlaceCard';

const style = {
  width: 330, margin: '0px auto',
};

const DndConteiner = ({
  nameItems, names, keys,
}) => {
  {
    const [cards, setCards] = useState(nameItems);
    if (nameItems.length !== cards.length) setCards(nameItems);

    const moveCard = (dragIndex, hoverIndex) => {
      const dragCard = cards[dragIndex];

      // making new order of names
      const newOrderNames = names;
      const moveName = newOrderNames.splice(dragIndex, 1);
      newOrderNames.splice(hoverIndex, 0, ...moveName);

      // making new order of keys
      const newOrderKeys = keys;
      const moveKey = newOrderKeys.splice(dragIndex, 1);
      newOrderKeys.splice(hoverIndex, 0, ...moveKey);

      setCards(
        update(cards, {
          $splice: [[dragIndex, 1], [hoverIndex, 0, dragCard]],
        }),
      );
    };

    return (
      <div style={style}>
        {cards.map((card, i) => (
          <PlaceCard
            key={card.id}
            index={i}
            id={card.id}
            contents={card.contents}
            moveCard={moveCard}
          />
        ))}
      </div>
    );
  }
};
export default DndConteiner;
