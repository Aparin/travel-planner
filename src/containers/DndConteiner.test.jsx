/* eslint-disable no-undef */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import HTML5Backend from 'react-dnd-html5-backend';
import { DragDropContextProvider } from 'react-dnd';
import DndConteiner from './DndConteiner';

const names = ['Moscow', 'Piter'];
const keys = [1, 2];
const item1 = (<div>Moscow</div>);
const item2 = (<div>Piter</div>);

const nameItems = [{ id: 1, contents: item1 }, { id: 2, contents: item2 }];

describe('DndConteiner', () => {
  it('renders DndConteiner without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <DragDropContextProvider backend={HTML5Backend}>
        <DndConteiner nameItems={nameItems} names={names} keys={keys} movePlace={() => {}} />
      </DragDropContextProvider>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  test('DndConteiner snapshot', () => {
    const component = renderer.create(
      <DragDropContextProvider backend={HTML5Backend}>
        <DndConteiner nameItems={nameItems} names={names} keys={keys} movePlace={jest.fn()} />
      </DragDropContextProvider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
