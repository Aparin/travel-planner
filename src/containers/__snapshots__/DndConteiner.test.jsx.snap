// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DndConteiner DndConteiner snapshot 1`] = `
<div
  style={
    Object {
      "margin": "0px auto",
      "width": 330,
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "border": "1px dashed gray",
        "cursor": "move",
        "display": "block",
        "margin": "5px auto",
        "opacity": 1,
        "padding": "5px",
      }
    }
  >
    <div>
      Moscow
    </div>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "border": "1px dashed gray",
        "cursor": "move",
        "display": "block",
        "margin": "5px auto",
        "opacity": 1,
        "padding": "5px",
      }
    }
  >
    <div>
      Piter
    </div>
  </div>
</div>
`;
